src/main/java/bio/terra/service/upgrade/Migrate.java

```diff
@@ -1,361 +1,59 @@
-
-            if (rs.next()) {
-                row = new DeploymentRow()
-                    .id(rs.getString("id"))
-                    .lockingPodName(rs.getString("locking_pod_name"));
-            }
-        } catch (PSQLException ex) {
-            if (!StringUtils.equals(ex.getSQLState(), PSQL_TABLE_NOT_EXIST)) {
-                throw new MigrateException("Select deployment failed", ex);
-            }
-        } catch (SQLException ex) {
-            throw new MigrateException("Select deployment failed", ex);
-        }
-
-        return row;
-    }
-
-    private void makeDeploymentTable() {
-        final String schemaCreate = "CREATE SCHEMA IF NOT EXISTS migrate";
-        final String tableCreate = "CREATE TABLE IF NOT EXISTS migrate.deployment_v1 (" +
-            " dep_version integer primary key," +
-            " id text," +
-            " locking_pod_name text)";
-
-        try (Connection connection = dataSource.getConnection();
-             PreparedStatement schemaStatement = connection.prepareStatement(schemaCreate);
-             PreparedStatement tableStatement = connection.prepareStatement(tableCreate)) {
-
-            startTransaction(connection);
-            schemaStatement.execute();
-            tableStatement.execute();
-            commitTransaction(connection);
-
-        } catch (SQLException ex) {
-            throw new MigrateException("Table create failed", ex);
-        }
-    }
-
-    private void migrateDatabase(String changesetFile, DataSource dataSource) {
-        try (Connection connection = dataSource.getConnection()) {
+    public void migrateDatabase() {
+        String changesetFile = dataRepoJdbcConfiguration.getChangesetFile();
+        try (Connection connection = dataRepoJdbcConfiguration.getDataSource().getConnection()) {
             Liquibase liquibase = new Liquibase(changesetFile,
                 new ClassLoaderResourceAccessor(),
                 new JdbcConnection(connection));
-            DatabaseChangeLogLock[] locks = liquibase.listLocks();
-            for (DatabaseChangeLogLock lock : locks) {
-                logger.info(String.format("dbChangeLogLock changeSet: %s, id: %s, lockedBy: %s, granted: %s",
-                    changesetFile, lock.getId(), lock.getLockedBy(), lock.getLockGranted()));
 
-                /*
-                 * We can get into this state where one of the APIs is running migrations and gets shut down so that
-                 * another API container can run. It will result in a lock that doesn't get released. This is similar
-                 * to the problems we will have from deploying multiple containers at once that try to run migrations.
-                 */
-                logger.warn("forcing lock release");
-                liquibase.forceReleaseLocks();
-            }
             logger.info(String.format("dropAllOnStart is set to %s", migrateConfiguration.getDropAllOnStart()));
             if (migrateConfiguration.getDropAllOnStart()) {
```

Comment:
The fact that this code path is now being run on every prod startup makes me a little nervous. Is there case where we'd allow dropall in prod? Maybe the dropall feature could be moved to code that cannot be run in prod, or is only run via another specific command.

Reply:
We could belt-and-suspenders it and add a spring profile check (basically, if it has the `terra` profile and not `alpha` or `staging`, then that's prod.

Also worth noting that this runs every time we migrate on prod today

Reply:
I think that's a good idea. Instead of a denylist I would make it an allowlist though, so only allowing `dropAllOnStart` when spring contains `testing` or `dev` . This way we need to explicitly opt-in the env to allow `dropAllOnStart` .

Reply:
Maybe this is another ticket and PR, if it turns out to be a lot of work.

Reply:
I added in the check, so let me know what you think! Also, I'm wondering if it should also include `integration` since it sets `DB_MIGRATE_DROPALLONSTART ` to `true`: https://github.com/broadinstitute/datarepo-helm-definitions/blob/master/integration/integration-1/datarepo-api.yaml#L15